### PR TITLE
[Example][Bugfix] graphsage node classification example

### PR DESCRIPTION
**Pull request by chang-l**
## Description

node_classification.py still fails due to the same issue https://github.com/dmlc/dgl/issues/4234.
This PR fixes node_classification.py runs under all modes (--mode = `cpu`, `mixed`, `puregpu`)
 
## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [x] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

## Changes

- test_idx always on cpu
- Accuracy calculation `MF.accuracy` always on cpu

### CODE REVIEW

**Review by yaox12 — APPROVED** [review]

File: examples/pytorch/graphsage/node_classification.py [hunk 1]
```diff
@@ -72,9 +72,9 @@ def evaluate(model, graph, dataloader):
 def layerwise_infer(device, graph, nid, model, batch_size):
     model.eval()
     with torch.no_grad():
-        pred = model.inference(graph, device, batch_size).to(device)
```
Comment (nv-dlasalle) [inline examples/pytorch/graphsage/node_classification.py:75]: Should we remove the `device` parameter then?

**Review by nv-dlasalle — APPROVED** [review]
Overall looks good to me, but want to verify that it works fine with all 3 modes: 'cpu', 'mixed', and 'puregpu'.

**Comment by chang-l** [comment]
> Overall looks good to me, but want to verify that it works fine with all 3 modes: 'cpu', 'mixed', and 'puregpu'.

Yes, it works with all 3 modes.

File: examples/pytorch/graphsage/node_classification.py (hunk 1 above)
Comment (chang-l) [reply examples/pytorch/graphsage/node_classification.py:75]: `device` is still needed to pass into `model.inference()` function.
